arm: fix thread's FPU initialization

agansari opened the pull request:
FPU's control and status register (FPCSR) for a thread is not initialized.
Random values are written to this register and this leads to failure on
fp_sharing test.
FPCSR register is set to 0 according to the value of FPDSCR.

Signed-off-by: Andrei Gansari <“andrei.gansari@nxp.com”>

Fixes: #14606 #9656

ioannisg reviewed (approved):
This fixes the failure of the FP sharing test in SAM E70 XPLAINED.

agansari committed:
arch: arm: fix thread's FPU initialization

FPU's control and status register (FPCSR) for a thread is not
initialized. Random values are written to this register and this leads
to failure on fp_sharing test.
FPCSR register is set to 0 according to the value of FPDSCR.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

MaureenHelm reviewed (approved):
nice catch
